code/datums/diseases/advance/symptoms/sensory.dm

```diff
@@ -30,21 +30,33 @@ Bonus
 		var/mob/living/M = A.affected_mob
 		switch(A.stage)
 			if(2)
-				if(M.reagents.get_reagent_amount("oculine")<10)
+				if(M.reagents.get_reagent_amount("oculine") < 10)
 					M.reagents.add_reagent("oculine", 10)
 					to_chat(M, "<span class='notice'>Your hearing feels clearer and crisp.</span>")
 			if(3)
-				if(M.reagents.get_reagent_amount("antihol") < 10 && M.reagents.get_reagent_amount("oculine") < 10 )
-					M.reagents.add_reagent_list(list("antihol"=10, "oculine"=10))
+				if(M.reagents.get_reagent_amount("antihol") < 10)
+					M.reagents.add_reagent("antihol" = 10)
```

Comment:
I'd make a proc from this method. One that takes the chemical, the max amount and the amount to add.
to make a proc here it'll be something like this:
/datum/symptom/sensory_restoration/match_max_reagent(var/reagent, var/max_amount, var/add_amount)

Reply:
And the syntax in this case is:
M.reagents.add_reagent(reagent,amount)
Like birdtalon said in his comment.